src/dex/verified/pools/secondary/secondarPool.ts

```diff
@@ -0,0 +1,400 @@
+import { BasePool } from '../../../balancer-v2/pools/balancer-v2-pool';
```

Comment:
@Mohzcrea8me secondary pool swaps work on an order book which matches orders based on price and amount, so you can't calculate price based on pool balance like in primary pool ? For secondary pool tokenIn and tokenOut, the calculation of price can be done by fetching pool data, extracting open orders and then matching orders which is what the SOR does, please check and confirm.

Reply:
Yes I will match it with sor, sor is wrong for primary when swapping security out it was (!iscashToken) instead of (iscashToken) the calculation from Solidity will be used too if  I notice any difference in result.